### PR TITLE
Replace whitelist/blacklist terminology with allowlist/denylist

### PR DESCRIPTION
**Pull request by thiagoalessio**
Inspired by:
https://github.com/rails/rails/pull/33681
https://github.com/elixir-lang/elixir/pull/10078

### CODE REVIEW

File: build-scripts/puddle-conf/atomic_openshift-3.0.conf [hunk 1]
```diff
@@ -13,4 +13,4 @@ emails = aos-devel@redhat.com,aos-qe@redhat.com
 announcer = openshift-puddle <aos-devel@redhat.com>
 publish = no
 keys = fd431d51,f21541eb
-blacklist = no
\ No newline at end of file
+denylist = no
```
Comment (sosiouxme) [inline build-scripts/puddle-conf/atomic_openshift-3.0.conf:16]: in puddle confs these are keywords.

of course, hopefully we won't need puddles much longer...

File: sjb/config/common/test_cases/origin_built_installed_release.yml [hunk 2]
```diff
@@ -45,7 +45,7 @@ extensions:
            wget "https://raw.githubusercontent.com/openshift/origin-ci-tool/master/oct/ansible/oct/callback_plugins/${plugin}.py"
            sudo mv "${plugin}.py" /usr/share/ansible/plugins/callback
         done
-        sudo sed -r -i -e 's/^#?stdout_callback.*/stdout_callback = default_with_output_lists/' -e 's/^#?callback_whitelist.*/callback_whitelist = generate_junit/' /etc/ansible/ansible.cfg
+        sudo sed -r -i -e 's/^#?stdout_callback.*/stdout_callback = default_with_output_lists/' -e 's/^#?callback_allowlist.*/callback_allowlist = generate_junit/' /etc/ansible/ansible.cfg
```
Comment (sosiouxme) [inline sjb/config/common/test_cases/origin_built_installed_release.yml:48]: these `sjb` entries look like they might have functional significance too... anyway, we don't usually mess with `sjb`

**Comment by vfreex** [comment]
@thiagoalessio I think we'd better to exclude the puddle configs and any files in sjb that we don't manage. What do you think?

**Comment by thiagoalessio** [comment]
initial attempt was a crude search&replace ... let me close this one and try substituting slowly, testing on `hack/` space, and open a new PR once I can verify everything works.

**Pull request closed without merging**